spec/services/status_update_service_spec.rb

```diff
@@ -1,159 +1,96 @@
 RSpec.describe StatusUpdateService do
-  let(:reference) { "b6589b2b-8f8e-457b-9ddf-237b62438ad1" }
-
-  let!(:delivery_attempt) do
-    create(:delivery_attempt, id: reference, status: "sending")
-  end
-
-  let(:status) { "delivered" }
-  let(:completed_at) { Time.zone.parse("2017-05-14T12:15:30.000000Z") }
-  let(:sent_at) { Time.zone.parse("2017-05-14T12:15:30.000000Z") }
-
-  subject(:status_update) { described_class.call(reference: reference, status: status, completed_at: completed_at, sent_at: sent_at) }
-
-  shared_examples "records successful stats" do
-    before { allow(GovukStatsd).to receive(:increment) }
-
-    it "records a success statistic" do
-      expect(GovukStatsd).to receive(:increment).with("status_update.success")
-      status_update
+  describe ".call" do
+    let!(:delivery_attempt) { create(:delivery_attempt, id: reference, email: email) }
+    let(:email) { create(:email) }
+    let(:reference) { SecureRandom.uuid }
+    let(:status) { "delivered" }
+    let(:completed_at) { Time.zone.now.beginning_of_minute }
+    let(:sent_at) { completed_at }
+    let(:user) { create(:user) }
+
+    let(:args) do
+      {
+        reference: reference,
+        status: status,
+        completed_at: completed_at,
+        sent_at: sent_at,
+        user: user,
+      }
     end
 
-    it "records a statistic on the number of status updates" do
-      expect(GovukStatsd).to receive(:increment).with("delivery_attempt.status.#{status.underscore}")
-      status_update
+    it "updates the delivery attempt record" do
```

Comment:
Do we need this if we're checking the individual statuses in (some of) the subsequent tests?

Reply:
This covers the behaviour that is consistent between statuses.